libs/components/src/lib/layout/layout.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { elementUpdated, fixture } from '@vivid-nx/shared';
+import { Size } from '../enums';
+import { AUTO_SIZING, Layout } from './layout';
+import '.';
+
+const COMPONENT_TAG = 'vwc-layout';
+
+describe('vwc-layout', () => {
+	let element: Layout;
+
+	beforeEach(async () => {
+		element = await fixture(`<${COMPONENT_TAG}>
+									<div style="border: 1px solid">test</div>
+									<div style="border: 1px solid">test</div>
+								</${COMPONENT_TAG}>`) as Layout;
+	});
+
+	describe('basic', () => {
+		it('initializes as a vwc-layout', async () => {
+			expect(element).toBeInstanceOf(Layout);
+			expect(element.gutters).toBeUndefined();
+			expect(element.columnBasis).toBeUndefined();
+			expect(element.columnSpacing).toBeUndefined();
+			expect(element.autoSizing).toBeUndefined();
+		});
+	});
+
+	describe('gutters', () => {
+		it('should set correct internal gutters style', async () => {
+			const gutters = Size.BaseLarge;
+			(element as any).gutters = gutters;
+			await elementUpdated(element);
+
+			const control = element.shadowRoot?.querySelector(`.control.gutters-${gutters}`);
+			expect(control)
+				.toBeInstanceOf(Element);
+		});
```

Comment:
Should I expect this to fail if I do not define `gutters`?
We'd might want to check "before and after".
```suggestion
			const gutters = Size.BaseLarge;
			const control = getControlElement();
			const classExistsWithoutGutters = control.classes.hasClass(`control.gutters-${gutters}`);
			(element as any).gutters = gutters;
			await elementUpdated(element);

			expect(classExistsWithoutGutters).toEqual(false);
			expect(control.classes.hasClass(`control.gutters-${gutters}`)).toEqual(true);
		});
```
You can even refactor for better readability by extracting to meaningfully named functions.